src/ripple/basics/impl/StringUtilities.cpp

```diff
@@ -89,11 +89,15 @@ uint64_t uintFromHex (std::string const& strSrc)
     return uValue;
 }
 
-// TODO Callers should be using beast::URL and beast::parse_URL instead.
 bool parseUrl (parsedURL& pUrl, std::string const& strUrl)
 {
     // scheme://username:password@hostname:port/rest
-    static boost::regex reUrl ("(?i)\\`\\s*([[:alpha:]][-+.[:alpha:][:digit:]]*)://([^/]+)(/.*)?\\s*?\\'");
+    static boost::regex reUrl (
+        "(?i)\\`\\s*"
+        "([[:alpha:]][-+.[:alpha:][:digit:]]*):"    //scheme
+        "//([^/]+)?"                                 // hostname
+        "(/.*)?"                                    // path and parameters
+        "\\s*?\\'");
```

Comment:
With the string split across multiple lines, using raw strings would look like this:
```
        R"regex((?i)\`\s*)regex"
        "([[:alpha:]][-+.[:alpha:][:digit:]]*):"    //scheme
        "//([^/]+)?"                                 // hostname
        "(/.*)?"                                    // path and parameters
        R"regex(\s*?\')regex");
```
I don't think that really helps readability. What do you think?

Reply:
I prefer the "classic" version, escape sequences and all.

Reply:
Using `R"regex(` instead of `R"(` clutters that up a bit - expecially for such a short string. But the backslashes aren't that bad. I'm fine with the "classic" version.

Reply:
Agreed, particularly since we're dealing with regex where `()` is so meaningful.